modules/gdscript/gdscript_compiler.cpp

```diff
@@ -2413,10 +2413,25 @@ Error GDScriptCompiler::_parse_class_level(GDScript *p_script, const GDScriptPar
 				const GDScriptParser::SignalNode *signal = member.signal;
 				StringName name = signal->identifier->name;
 
-				Vector<StringName> parameters_names;
+				Vector<Pair<StringName, Variant::Type>> parameters_names;
 				parameters_names.resize(signal->parameters.size());
 				for (int j = 0; j < signal->parameters.size(); j++) {
-					parameters_names.write[j] = signal->parameters[j]->identifier->name;
+					const GDScriptParser::ParameterNode &this_parameter = *member.signal->parameters[j];
+					// has signal_arg_type?
+					Variant::Type type = Variant::NIL;
+					if (this_parameter.datatype_specifier && this_parameter.datatype_specifier->type_chain.size() == 1) {
+						const Variant::Type builtin_type = GDScriptParser::get_builtin_type(this_parameter.datatype_specifier->type_chain[0]->name);
+						// is builtin ?
+						// todo: If in `GDScriptParser::parse_signal()` it can be assumed that the type is always a built-in type then the following test is not required.
```

Comment:
```suggestion
						// TODO: If in `GDScriptParser::parse_signal()` it can be assumed that the type is always a built-in type, then the following test is not required.
```